src/kibali/AuthZChecker.cs

```diff
@@ -18,6 +18,10 @@ public class AuthZChecker
         private readonly Dictionary<string, ProtectedResource> resources = new Dictionary<string, ProtectedResource>();
         private OpenApiUrlTreeNode urlTree;
 
+        public HashSet<PermissionsError> Errors = new HashSet<PermissionsError>();
```

Comment:
I'm not a big fan of attaching the errors to the AuthZChecker and ProtectedResource object.  How would you feel about returning an errors collection from the Validate method?

Reply:
Makes sense. Updating.